src/pf/parse.lua

```diff
@@ -24,12 +24,55 @@ local punctuation = set(
    '+', '-', '*', '/', '&', '|', '^', '&&', '||', '<<', '>>', '\\'
 )
 
+local number_terminators = " \t\r\n)]:!<>=+-*/%&|^"
+
+local function lex_number(str, pos, base)
+   local res = 0
+   local i = pos
+   while i <= #str do
+      local chr = str:sub(i,i)
+      local n = tonumber(chr, base)
+      if n then
+         res = res * base + n
+         assert(res <= 0xffffffff, 'integer too large: '..res)
```

Comment:
Yep. Adding more 'f's in the middle retains validity as a domain name, so I'm confirming your observation.